support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -162,27 +175,51 @@ object ZuoraSubscriptionCreator {
     }
   }
 
+  // scalastyle:off cyclomatic.complexity
   def getEmailState(
     state: CreateZuoraSubscriptionState,
     accountNumber: ZuoraAccountNumber,
     subscriptionNumber: ZuoraSubscriptionNumber,
     paymentOrRedemptionData: Either[PaymentMethodWithSchedule, RedemptionData],
-    maybeGiftPurchase: Option[GiftPurchase],
-  ): SendThankYouEmailState =
+    maybeDigitalSubscriptionGiftCreationDetails: Option[DigitalSubscriptionGiftCreationDetails],
+  ): SendThankYouEmailState = {
+    val Purchase = Left
+    type Redemption = Right[PaymentMethodWithSchedule, RedemptionData]
     SendThankYouEmailState(
       requestId = state.requestId,
       user = state.user,
-      giftPurchase = maybeGiftPurchase,
-      product = state.product,
       analyticsInfo = state.analyticsInfo,
-      paymentOrRedemptionData = paymentOrRedemptionData,
-      firstDeliveryDate = state.firstDeliveryDate,
-      promoCode = state.promoCode,
+      productTypeCreated = paymentOrRedemptionData match {
```

Comment:
This is quite nasty, I think you need to break the nested matches out into separate functions if we're going to go down this route

Reply:
good point and even scalastyle was complaining about it